lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -2287,15 +2289,25 @@ namespace Js
             case UDAT_DOW_LOCAL_FIELD:
                 typeString = _u("weekday"); break;
             case UDAT_AM_PM_FIELD:
+            case UDAT_AM_PM_MIDNIGHT_NOON_FIELD:
```

Comment:
It looks like this was added in ICU v57, so you will probably have issues on Ubuntu 16.04 w/ ICU 55.